### PR TITLE
Add KMS decryption for sensitive environment variables

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

File: amplitude.py [hunk 1]
```diff
@@ -20,18 +23,40 @@
...
+
     for record in records:
         if record["eventSource"] != "aws:s3":
             continue
 
+        print record["s3"]["bucket"]["name"], record["s3"]["object"]["key"]
         s3 = boto3.resource("s3", region_name=record["awsRegion"])
         s3_object = s3.Object(record["s3"]["bucket"]["name"], record["s3"]["object"]["key"])
+        tmpfileobj, tmppath = tempfile.mkstemp()
+        s3_object.download_file(tmppath)
```
Comment (philbooth) [inline amplitude.py]: Did you do this because the streaming approach doesn't work? If that's the case it's probably worth us spending a little bit of time trying to make it work.

And if we can't make it work, we should make this change as a separate commit so it's visible in the history.

**Comment by jbuck** [comment]
I've split up this PR into 3 and it's ready for review now

File: amplitude.py [hunk 2]
```diff
@@ -20,6 +22,21 @@
...
+    return kms_decrypt(b64decode(os.environ[key]))
+
+def kms_decrypt(encrypted_data):
+    """Decrypt KMS variables"""
+    res = boto3.client("kms").decrypt(
+        CiphertextBlob=encrypted_data,
+    )
+    return res["Plaintext"].decode("utf-8")
+
+if os.environ["LAMBDA_TASK_ROOT"]:
```
Comment (philbooth) [inline amplitude.py]: This raises a `KeyError` if `LAMBDA_TASK_ROOT` is not defined. We should avoid that so we can still test locally:

```py
if "LAMBDA_TASK_ROOT" in os.environ:
```
